scripts/scrub-user.js

```diff
@@ -0,0 +1,195 @@
+  const client_ids = results.map(each => each.client_id);
+
+  const summary = {
+    email: email,
+    length: results.length,
+    ids: client_ids,
+    clip_total: clip_total,
+    first_clip: null,
+    in_past_dataset: false,
+  };
+
+  if (summary.clip_total) {
+    summary.first_clip = results[0].first_clip;
+
+    if (LAST_DATASET >= new Date(summary.first_clip))
+      summary.in_past_dataset = true;
+  }
+
+  return summary;
+};
+
+const processIteration = last => {
+  console.log(`=======================================`);
+  safeToDelete = true;
+
+  if (last) {
+    printSummary(deletionSummary);
+    readline.close();
+    process.exit(1);
+  }
+};
+
+const printSummary = summary => {
+  console.log('Deletion summary: \n');
+  if (deletionSummary.emails.length)
+    console.log(
+      `Removed user information related to ${deletionSummary.emails.join(', ')}`
+    );
+
+  if (deletionSummary.s3_ids.length)
+    console.log(
+      `Please manually remove the following folders from S3: ${deletionSummary.s3_ids.join(
+        ', '
+      )}.`
+    );
+
+  if (deletionSummary.dataset_ids.length)
+    console.log(
+      `The following client_ids were included in a past dataset: ${deletionSummary.dataset_ids.join(
+        ', '
+      )}.`
+    );
+};
+
+const printUserStats = stats => {
+  console.log(`
+    Email: ${stats.email}
+    Associated client_ids: ${stats.length}
+    Total clips contributed: ${stats.clip_total}
+    First contribution: ${stats.first_clip}
+    `);
+};
+
+const deleteClipRecords = (connection, userStats, isLast) => {
+  const idStrings = userStats.ids.map(id => `"${id}"`);
+  connection.query(
+    `DELETE FROM clips WHERE client_id IN (${idStrings})`,
+    (clip_error, clip_results, clip_fields) => {
+      if (clip_error) throw clip_error;
+
+      if (clip_results.affectedRows) {
+        deletionSummary.s3_ids = deletionSummary.s3_ids.concat(userStats.ids);
+
+        if (userStats.in_past_dataset)
+          deletionSummary.dataset_ids = deletionSummary.dataset_ids.concat(
+            userStats.ids
+          );
+
+        console.log(
+          `Records for ${clip_results.affectedRows} clips were deleted.\n`
+        );
+      }
+
+      processIteration(isLast);
+    }
+  );
+};
+
+const deleteUserRecords = (connection, userStats, isLast) => {
+  connection.query(
+    `DELETE FROM user_clients WHERE email = ${userStats.email}`,
+    (delete_error, delete_results, delete_fields) => {
+      if (delete_error) throw delete_error;
+
+      deletionSummary.emails.push(userStats.email);
+
+      if (userStats.clip_total) {
+        readline.question(
+          `\nWould you also like to delete the ${userStats.clip_total} associated clips? \nThis CANNOT be reversed [Y/N] `,
+          answer => {
+            if (answer.toLowerCase() === 'y')
+              deleteClipRecords(connection, userStats, isLast);
+            else processIteration(isLast);
```

Comment:
If you wanted to account for typos, you could create a simple loop like:

```javascript
function readlineLoop (prompt, options) {
  readline.question(prompt, answer => {
    const callback = options[answer.toLowerCase()];
    if (callback) callback();
    else readlineLoop(prompt, options);
  });
}
```

then call it with:

```javascript
readlineLoop(`\nWould you also like to delete the ${userStats.clip_total} associated clips? \nThis CANNOT be reversed [Y/N] `, {
  y: () => deleteClipRecords(connection, userStats, isLast),
  n: () => processIteration(isLast),
});
```

Reply:
Oooh this is neat and elegant, thank you!